index.js

```diff
@@ -210,9 +211,17 @@ function validateSignature(author, payload, signature, hmacKey) {
   const hmacKeyErr = validateHmacKey(hmacKey)
   if (hmacKeyErr) return hmacKeyErr
 
-  if (!ssbKeys.verify(author, signature, hmacKey, payload))
+  if (
+    !ssbKeys.verify(
+      { public: author, curve: 'ed25519' },
+      signature,
+      hmacKey,
+      payload
+    )
+  )
     return new Error(
-      'invalid message: signature must correctly sign the payload'
+      'invalid message: signature must correctly sign the payload',
+      author
```

Comment:
I was initially a bit confused what signature was the problem here, so this was to help me out. There are the outer and contentSignature. The contentSignature you can only validate if the content is not encrypted which is why I didn't want to include validation here but at a higher level (ssb-meta-feeds module). Come to think of it, it might not be such a bad idea to validate it if we can here. What do you think? Note the contentSignature is a big different because the signed content is concatted with bendybutt :)

Reply:
The `author` data is a nice addition!

I don't have a strong opinion on where validation for `content` should live. Considering that the `contentSignature` is generated by this module (https://github.com/ssb-ngi-pointer/ssb-bendy-butt/blob/master/index.js#L111), perhaps it's best to have validation here too? Are there any other validation checks required for `content`, besides the signature?